runelite-client/src/main/java/net/runelite/client/plugins/performance/PerformanceOverlay.java

```diff
@@ -84,21 +86,62 @@ private Color getFpsValueColor()
 		return isEnforced() ? Color.red : Color.yellow;
 	}
 
+	private static Color getPingColor(int ping)
+	{
+		if (ping >= 100 || ping < 0)
+		{
+			return Color.red;
+		}
+		else if (ping >= 50)
+		{
+			return Color.yellow;
+		}
+		return Color.green;
+	}
+
+	private int calculateOffset()
+	{
+		if ((client.getVar(Varbits.SIDE_PANELS) == 1) && client.isResized())
+		{
+			return 27;
+		}
+
+		return 2;
+	}
+
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
-		if (!config.drawFps())
+		if (!config.drawFps() && !config.drawPing())
 		{
 			return null;
 		}
-		
-		final String text = client.getFPS() + FPS_STRING;
-		final int textWidth = graphics.getFontMetrics().stringWidth(text);
-		final int textHeight = graphics.getFontMetrics().getAscent() - graphics.getFontMetrics().getDescent();
 
+		final int offset = calculateOffset();
 		final int width = (int) client.getRealDimensions().getWidth();
-		final Point point = new Point(width - textWidth - VALUE_X_OFFSET, textHeight + Y_OFFSET);
-		OverlayUtil.renderTextLocation(graphics, point, text, getFpsValueColor());
+		final FontMetrics fontMetrics = graphics.getFontMetrics();
+		final int textHeight = fontMetrics.getAscent() - fontMetrics.getDescent();
+
+		if (config.drawFps())
+		{
+			final int fps = client.getFPS();
+			final String fpsText = String.format("%d FPS", fps);
+			final int textWidth = fontMetrics.stringWidth(fpsText);
+
+			Point point = new Point(width - textWidth - offset, Y_OFFSET + textHeight);
+			OverlayUtil.renderTextLocation(graphics, point, fpsText, getFpsValueColor());
+		}
+
+		if (config.drawPing())
+		{
+			// Shift up when FPS is not being drawn
+			final int pingYOffset = config.drawFps() ? 11 : 0;
```

Comment:
You can probably just add to the offset (e.g, set base offset, increment it if FPS is drawn, use it here)

Reply:
Good suggestion, I've changed it a bit and got rid of `textHeight` as well